### PR TITLE
add seed and filter-options to rdock, terminate if job gets stuck in a loop

### PR DESCRIPTION
**Pull request by bgruening**

### CODE REVIEW

File: chemicaltoolbox/rdock/rbdock.xml [hunk 1]
```diff
@@ -13,15 +13,33 @@ ln -s $receptor_prm receptor.prm &&
...
+       -r receptor.prm
+       -p dock.prm
+       -n $num
+       -o output
+       #if str($seed):
+        -s $seed
+       #end if
+&&
+
+
```
Comment (nsoranzo) [inline chemicaltoolbox/rdock/rbdock.xml:27]:
```suggestion
```

File: chemicaltoolbox/rdock/rbdock.xml [hunk 2]
```diff
@@ -13,15 +13,33 @@ ln -s $receptor_prm receptor.prm &&
...
+sdsort -n -s -fSCORE output.sd 
+
+#if $filter.filter_select == "filter":
+    |
```
Comment (nsoranzo) [inline chemicaltoolbox/rdock/rbdock.xml:31]:
```suggestion
```

File: chemicaltoolbox/rdock/rbdock.xml [hunk 3]
```diff
@@ -13,15 +13,33 @@ ln -s $receptor_prm receptor.prm &&
...
+    #if $filter.score:
+      sdfilter -f'\$SCORE <= $filter.score' |
```
Comment (nsoranzo) [inline chemicaltoolbox/rdock/rbdock.xml:33]:
```suggestion
        | sdfilter -f'\$SCORE <= ${filter.score}'
```

File: chemicaltoolbox/rdock/rbdock.xml [hunk 4]
```diff
@@ -13,15 +13,33 @@ ln -s $receptor_prm receptor.prm &&
...
+    #end if
+    #if $filter.nscore:
+      sdfilter -f'\$SCORE.norm <= $filter.nscore' |
```
Comment (nsoranzo) [inline chemicaltoolbox/rdock/rbdock.xml:36]:
```suggestion
        | sdfilter -f'\$SCORE.norm <= ${filter.nscore}'
```

File: chemicaltoolbox/rdock/rbdock.xml [hunk 5]
```diff
@@ -13,15 +13,33 @@ ln -s $receptor_prm receptor.prm &&
...
+    #end if
+    sdfilter -f'\$_COUNT <= $filter.top'
```
Comment (nsoranzo) [inline chemicaltoolbox/rdock/rbdock.xml:38]:
```suggestion
    | sdfilter -f'\$_COUNT <= $filter.top'
```

**Comment by bgruening** [comment]
Thanks @nsoranzo ... we had some problems with molecules that do not terminate. There is now a test that should catch this.

**Comment by bgruening** [comment]
Arg, this seems to work with 19.09, but not with 20.01 ...

**Comment by simonbray** [comment]
What about the idea of wrapping rbdock in the python script? Or is this overkill?

Really don't get why this doesn't work. Just out of curiosity, I'll check my grep solution with Galaxy 19.09 and see if it works.

**Comment by simonbray** [comment]
> What about the idea of wrapping rbdock in the python script?

Have written something for this, just testing it now.

**Comment by simonbray** [comment]
Okay, whether this is a good solution is very debatable but tests are passing for me locally.

**Comment by simonbray** [comment]
I'm okay to merge I think, what do you think @bgruening @tdudgeon?

File: chemicaltoolbox/rdock/rbdock.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import subprocess
+import argparse
+
+def main():
+    parser = argparse.ArgumentParser(description='Simple wrapper for rbdock')
+    parser.add_argument('-n', '--num', type=int, help='Equilibrated system as input')
```
Comment (tdudgeon) [inline chemicaltoolbox/rdock/rbdock.py:6]: Description needs updating?

**Comment by tdudgeon** [comment]
I think we'll need to test this carefully, but let's go for it and deploy?

File: chemicaltoolbox/rdock/rbdock.py (hunk 6 above)
Comment (simonbray) [reply chemicaltoolbox/rdock/rbdock.py]: Oh yes, thanks for the catch

**Pull request merged**